### ISSUE TITLE
make serverPrefetch() rejection trappable

### ISSUE THREAD

**Issue by FranckFreiburger** [vuejs/vue#9539]
### What problem does this feature solve?
Currently, rejections in `serverPrefetch` cannot be handled
eg.
```
serverPrefetch() {

   return Promise.reject('myError')
}
```

### What does the proposed API look like?
Maybe send `serverPrefetch` rejections as `renderStream` error event ?

eg.

```
const renderStream = renderer.renderToStream(context)
renderStream.on('error', err => { ... })
```

**Comment by skyline0705** [comment]
Is there necessary to stop the renderer and throw an error for renderStream?
In Vue 2.6.x, I thinks this means "only this component has error, not the hold app",so it choose to stop render this component only, and the warning will be report when hydrating in browser.

**Comment by FranckFreiburger** [comment]
I agree that stopping the rendering is too severe for a rejection, but we need at least a mean to accurately detect that something is going wrong during a `serverPrefetch`.

**Comment by CyberAP** [comment]
Just came across this. This issue actually prevents me from implementing `renderToStream` in my app at all because there's just no way of recovering from that state and for example refresh the page using `renderToString`. Really hope this could be fixed in Vue 2.7 and we'd get an error event if serverPrefetch fails.
